lua/neo-tree/utils.lua

```diff
@@ -246,11 +246,13 @@ end
 M.get_modified_buffers = function()
   local modified_buffers = {}
   for _, buffer in ipairs(vim.api.nvim_list_bufs()) do
-    local buffer_name = vim.api.nvim_buf_get_name(buffer)
-    if buffer_name == nil or buffer_name == "" then
-      buffer_name = "[No Name]#" .. buffer
+    if vim.api.nvim_buf_is_loaded(buffer) and vim.fn.buflisted(buffer) then
```

Comment:
Modified buffers should be loaded and listed, so I added this instead of creating a new method to check for open files to avoid code repetition.